joomla/joomla-cms
# Use Accredited Languagepack XML directly to show the installable languages

Currently, the "Install Languages" view of the Extension Installer shows only non-installed languages to choose from. This makes sense to a degree since there seems to be no need to install already installed languages.
However in some cases you may want to do exactly that. Eg if you installed languages using discover you miss the package manifest and want to reinstall the language properly to get that as well. Or you may have accidentally deleted some files and want to reinstall the language to fix that.

### Summary of Changes
This changes the view from using the #__updates table to directly read from the XML manifest file. Currently it reads that file (https://update.joomla.org/language/translationlist_3.xml), parses it for available updates (not installed languages count as update as well!) and stores that into the #__updates table, then afterwards reads that table and compiles the list of installable languages.
With this PR we skip the part with the #__updates table and directly use the content of the file so we have also the already installed languages.

### Testing Instructions
* Make sure you can install any languages as expected.
* Already installed languages will show a "Reinstall" button instead of the "Install" button.

### Sidenote
Due to the missing underlying database table, we can't use the IDs anymore to install multiple languages at once and thus the button "Install" has been removed. The other button "Find Languages" also isn't needed anymore as the list is taken directly from the XML file now.

### Documentation Changes Required
This may need adjustments in user documentation (docs pages, tutorials?) due to the changed view layout.
